File: backend/src/main/java/com/programmers/pcquotation/global/security/CustomUserDetails.java
package com.programmers.pcquotation.global.security;

import java.util.Collection;
import java.util.Collections;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import com.programmers.pcquotation.domain.customer.service.CustomerService;
import com.programmers.pcquotation.domain.member.entitiy.Member;
import com.programmers.pcquotation.domain.member.service.AuthService;
import com.programmers.pcquotation.domain.seller.service.SellerService;
import com.programmers.pcquotation.global.enums.UserType;

import lombok.RequiredArgsConstructor;


public class CustomUserDetails implements UserDetails {
	private final Member member;

	// Constructor
	public CustomUserDetails(Member member) {
		this.member = member;
	}

	@Override
	public Collection<? extends GrantedAuthority> getAuthorities() {
		return member.getAuthorities();
	}

	public Long getId() {
		return member.getId();
	}



	@Override
	public String getUsername() {
		return member.getUsername();
	}

	@Override
	public String getPassword() {
		return member.getPassword();
	}

}

File: backend/src/main/java/com/programmers/pcquotation/global/security/CustomUserDetails.kt
package com.programmers.pcquotation.global.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import com.programmers.pcquotation.domain.member.entitiy.Member

class CustomUserDetails(
    private val member: Member
) : UserDetails {

    val id: Long
        get() = member.id ?: throw IllegalStateException("Id cannot be null")

    override fun getAuthorities(): Collection<out GrantedAuthority> =
        member.authorities ?: throw IllegalStateException("Authorities cannot be null")

    override fun getUsername(): String = member.username ?: throw IllegalStateException("Username cannot be null")

    override fun getPassword(): String = member.password ?: throw IllegalStateException("password cannot be null")

    override fun isAccountNonExpired(): Boolean = true

    override fun isAccountNonLocked(): Boolean = true

    override fun isCredentialsNonExpired(): Boolean = true

    override fun isEnabled(): Boolean = true
}
